File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i : tangerine){
           map.put(i, map.getOrDefault(i, 0) + 1);
        }
        List<Integer> list=new ArrayList<>(map.values());
        Collections.sort(list, Collections.reverseOrder());
        for(int j : list){
            k -= j;
            answer++;
            if(k < 1) break;
        }
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    dict = {}
    for i in range(len(tangerine)):
        dict[tangerine[i]] = dict.get(tangerine[i], 0) + 1
        
    sort = sorted(dict.values(), reverse=True)
    
    for j in range(len(sort)):
        answer += 1
        k -= sort[j]
        if k <= 0:
            break
    
    return answer